controllers/spiaccesstoken_controller.go

```diff
@@ -166,21 +166,22 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
-	// cleanup tokens which are in awaiting state and have their parent binging removed.
-	// grace period taken in account to avoid race condition during the object creation process (i.e. when token is created but not yet linked)
-	if at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData && time.Since(at.CreationTimestamp.Time).Seconds() > NoLinkingBindingGracePeriodSeconds {
+	tokenLifetime := time.Since(at.CreationTimestamp.Time).Seconds()
+
+	// cleanup tokens by lifetime or on being unreferenced by any binding in the AwaitingToken state
+	if (tokenLifetime > r.Configuration.AccessTokenTtl.Seconds()) || (at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData && tokenLifetime > GracePeriodSeconds) {
 		hasLinkedBindings, err := hasLinkedBindings(ctx, &at, r.Client)
 		if err != nil {
-			lg.Error(err, "failed to validate the object", "token", at.ObjectMeta.Name, "error", err)
-			return ctrl.Result{}, fmt.Errorf("failed to validate the object: %w", err)
+			lg.Error(err, "failed to check linked bindings for token", "token", at.ObjectMeta.Name, "error", err)
```

Comment:
no need to add `token` to the logger, because controller runtime already added the value for it to the logger in the context.